src/components/controls/color-by.js

```diff
@@ -69,7 +69,7 @@ class ColorBy extends React.Component {
   }
 
   // State from the outside world enters via props.
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
Does this `UNSAFE_…` naming work on React 16?

Reply:
Should work [from 16.3 onwards](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#gradual-migration-path)

Reply:
Also, I assume that CI should be able to catch if it doesn't work, since Auspice pins React 16.8.6.

Reply:
Thanks both!

I tried briefly to find the changelog/docs for the `UNSAFE_…` prefixing and came up empty, but didn't then expand to a general search that would have found the blog post.

@victorlin Ah, it wasn't clear to me that CI here would catch if these method names would actually get used, i.e. I wasn't sure if CI runs Auspice in a live browser environment and if tests would break if these were not called.

Reply:
I take back my comment – I just tried renaming the method to something random and none of the npm scripts run in CI produced any errors.

It looks like `eslint-plugin-react` [automatically detects](https://github.com/jsx-eslint/eslint-plugin-react/blob/4a92667e4b99221013e1d2aa61e97296895cecc1/README.md?plain=1#L43-L45) the installed version of React, however specifying `settings.react.version: "16.0"` doesn't produce any "unused method" messages.

This requires some manual testing.

Reply:
I do note this warning from eslint-plugin-react when running linting:

    Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .

Reply:
Oh, I misread the snippet linked above. Pushed 4adfc9b275a38d19fe5f144477c4720a881ce9f0 to address the warning.

Reply:
Added manual testing notes for `UNSAFE_*` methods in the PR description.